src/basic/macro.h

```diff
@@ -506,4 +506,21 @@ static inline int __coverity_check__(int condition) {
         DEFINE_PUBLIC_TRIVIAL_REF_FUNC(type, name);                    \
         DEFINE_PUBLIC_TRIVIAL_UNREF_FUNC(type, name, free_func);
 
+/* Negative return values from impl are mapped to EXIT_FAILURE, and
+ * everything else means success! */
+#define DEFINE_MAIN_FUNCTION(impl)                                      \
+        int main(int argc, char *argv[]) {                              \
+                int r = impl(argc, argv);                               \
```

Comment:
hmm, according to CODING_STYLE it's not cool to mix variable declarations and function invocations in one line